packages/suite/src/components/suite/AppNavigationPanel/index.tsx

```diff
@@ -102,42 +21,31 @@ interface Props {
     maxWidth: 'small' | 'default';
     children?: React.ReactNode;
     navigation?: React.ReactNode;
-    navigationSticky?: React.ReactNode;
 }
 
+export const isInView = () => React.useContext(AppNavigationPanelCtx);
```

Comment:
You are creating custom hook here so it should have a name starting with `use...` to be properly checked for violations of rules of Hooks.

see lint error here: https://gitlab.com/satoshilabs/trezor/trezor-suite/-/jobs/1402987761
> React Hook "React.useContext" is called in function "isInView" that is neither a React function component nor a custom React Hook function.

The message is a bit confusing, because isInView actually IS a hook.

Let's rename it 
```suggestion
export const useInView = () => React.useContext(AppNavigationPanelCtx);
```

(side note: it would be easier in this case just pass the additional prop to component in `props.navigation` directly and avoid using context, but your solution is more flexible so leave it like that, just fix the name...)

Reply:
useInView already exists in the project (comes from intersection observer lib), changed to useInViewProp instead in 331a45e053a53457e46506ccff18f29dd5f5b5fc